packages/tangle/message.go

```diff
@@ -1,56 +1,81 @@
 package tangle
 
 import (
+	"bytes"
 	"fmt"
+	"math/bits"
+	"sort"
 	"sync"
 	"time"
 
 	"github.com/iotaledger/goshimmer/packages/tangle/payload"
+	"github.com/iotaledger/hive.go/bitmask"
+	"github.com/iotaledger/hive.go/cerrors"
 	"github.com/iotaledger/hive.go/crypto/ed25519"
 	"github.com/iotaledger/hive.go/marshalutil"
 	"github.com/iotaledger/hive.go/objectstorage"
 	"github.com/iotaledger/hive.go/stringify"
+	"github.com/iotaledger/hive.go/types"
 	"github.com/mr-tron/base58"
 	"golang.org/x/crypto/blake2b"
+	"golang.org/x/xerrors"
 )
 
 const (
+	// MessageVersion defines the version of the message structure.
+	MessageVersion uint8 = 1
+
 	// MaxMessageSize defines the maximum size of a message.
 	MaxMessageSize = 64 * 1024
 
 	// MessageIDLength defines the length of an MessageID.
-	MessageIDLength = 64
+	MessageIDLength = 32
 
-	// ContentIDLength contains the amount of bytes that a marshaled version of the ContentID contains.
-	ContentIDLength = MessageIDLength
-)
+	// StrongParent identifies a strong parent in the bitmask.
+	StrongParent uint8 = 1
+
+	// WeakParent identifies a weak parent in the bitmask.
+	WeakParent uint8 = 0
 
-// ContentID identifies the content of a message without its parent1/parent2 ids.
-type ContentID = MessageID
+	// MinParentsCount defines the minimum number of parents a message must have.
+	MinParentsCount = 1
```

Comment:
Should this not be `2`?

Reply:
we decided to make it `1` because you can't get `2` unique parents if there is only one tip, for example during bootstrapping the tangle. In current implementation, you just have `parent1` = `parent2` in these cases, which is equivalent to have 1 strong parent.